internal/backend/message_tx.go

```diff
@@ -359,3 +359,30 @@ func txUpdateMessageID(ctx context.Context, tx *ent.Tx, oldID, newID string) err
 
 	return nil
 }
+
+func txMarkMessageAsDeleted(ctx context.Context, tx *ent.Tx, messageID string) error {
+	if _, err := tx.Message.Update().Where(message.MessageID(messageID)).SetDeleted(true).Save(ctx); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func txDeleteMessagesMarkedDeleted(ctx context.Context, tx *ent.Tx, messageIDs ...string) error {
```

Comment:
It took me a while to figure out why these functions have the signature they do, and i'm not a fan tbh. If i call something called `txDeleteMessagesMarkedDeleted`, i expect it to do exactly that, not limited just to a set of messages i then pass in.

Why complicate things? I'd just have a getter (`txGetMessagesPendingDeletion`) and setter (`txSetMessagePendingDeletion`) for the `Deleted` field, as well as a method for removing rows in the `Messages` table by ID (`txDeleteMessages`). Much simpler imo.

Reply:
We now only have the function `txDeleteMessages`